onnxruntime/test/onnx/testcase_request.h

```diff
@@ -0,0 +1,116 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include "callables.h"
+#include "TestCaseResult.h"
+#include "test_allocator.h"
+#include <core/platform/env_time.h>
+#include "core/platform/threadpool.h"
+#include <core/session/onnxruntime_cxx_api.h>
+
+#include <atomic>
+#include <memory>
+#include <mutex>
+#include <condition_variable>
+
+class ITestCase;
+
+namespace onnxruntime {
+
+namespace concurrency {
+class ThreadPool;
+}
+
+using PThreadPool = onnxruntime::concurrency::ThreadPool*;
+
+namespace test {
+
+/// <summary>
+/// Runs a single Test Case (model)
+/// </summary>
+class TestCaseRequestContext {
+ public:
+  using Callback = Callable<void, std::shared_ptr<TestCaseResult>>;
+
+  /// <summary>
+  /// Runs data tests on the model sequentially (concurrent_runs < 2)
+  ///  and repeats them (repeat_count > 1) or concurrently (never repeats)
```

Comment:
nit: Right now someone can set both repeat > 1 and concurrent_runs > 1. May make sense to document this in the usage - when concurrent_runs > 1 repeat is ignored

Reply:
I thought this is what the comment says.

---
In reply to: [489080098](https://github.com/microsoft/onnxruntime/pull/5169#discussion_r489080098) [](ancestors = 489080098)

Reply:
Comment here is clear. I was referring to updating the usage for test runner to state the same.